sys/vm_map.c

```diff
@@ -181,6 +181,18 @@ int vm_map_findspace(vm_map_t *map, vm_addr_t start, size_t length,
   return 0;
 }
 
+int vm_map_expand(vm_map_t *map, vm_map_entry_t *entry, vm_addr_t new_end) {
```

Comment:
I think `sbrk` can force a `vm_map_entry` to shrink if passed negative argument. Thus I suggest to rename this routine to `vm_map_entry_resize` and handle `new_end < entry->end` case.